dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/category/CategoryComboDeletionHandler.java

```diff
@@ -58,8 +59,10 @@ private DeletionVeto allowDeleteCategory(Category category) {
   }
 
   private void deleteCategoryOptionCombo(CategoryOptionCombo categoryOptionCombo) {
-    for (CategoryCombo categoryCombo : categoryService.getAllCategoryCombos()) {
```

Comment:
FYI: I am not sure why we would loop over all CCs when there is just one linked to the COC which needs unlinking. Also I added a check if the un-linking isn't already done because if this was called as a consequence of deleting the CC it has and so the update should not be called again on the CC.

Reply:
If not needed, this is very nice. That could be very costly.
An existing/new test would confirm that.

Reply:
yeah, I only have my subjective observation from test and debugging but I found this because an deletion attempt would just spin my PC out of control so I had to kill the server, then step through it to see what was going on, which was when I found this. I then redid the deletion and it still was super slow but the server was at least responsive.